### PR TITLE
Bug/pn 7923

### PR DESCRIPTION
**Pull request by ariannazafarana**

### CODE REVIEW

File: src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java [hunk 1]
```diff
@@ -114,6 +83,39 @@ public void execute(String operationId) {
                 .block();
     }
 
+    private Mono<Void> checkValidationFlow(PnServiceDeskOperations operation, PnServiceDeskAddress addres) {
+        return getIuns(operation.getRecipientInternalId())
+                .collectList()
+                .doOnNext(responsePaperNotificationFailed -> {
```
Comment (vincenzo-racca-pa) [inline src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java:89]: utilizziamo il flatMap invece che il doOnNext in questi casi:

.flatMap(responsePaperNotificationFailed -> {
log.debug("listOfIuns = {}, List of iuns retrivied", responsePaperNotificationFailed);
operation.setErrorReason(null);
log.debug("errorReason = {}, Setting to null errorReason into entityOperation", operation);
return updateOperationStatus(operation, OperationStatusEnum.VALIDATION).thenReturn(responsePaperNotificationFailed);
})

File: src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java [hunk 2]
```diff
@@ -114,6 +83,39 @@ public void execute(String operationId) {
...
+                    operation.setErrorReason(null);
+                    log.debug("errorReason = {}, Setting to null errorReason into entityOperation", operation);
+                    updateOperationStatus(operation, OperationStatusEnum.VALIDATION);
+                })
+                .flatMapMany(Flux::fromIterable)
+                .flatMap(iun -> {
+                    log.debug("iun = {}, Get attachment from iun", iun);
+                    return getAttachmentsFromIun(operation, iun);
+                })
+                .flatMap(pnServiceDeskAttachments -> getFileKeyFromAttachments(pnServiceDeskAttachments))
```
Comment (vincenzo-racca-pa) [inline src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java:100]: utilizziamo il method reference: .flatMap(this::getFileKeyFromAttachments)

File: src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java [hunk 3]
```diff
@@ -114,6 +83,39 @@ public void execute(String operationId) {
                 .block();
     }
 
+    private Mono<Void> checkValidationFlow(PnServiceDeskOperations operation, PnServiceDeskAddress addres) {
```
Comment (vincenzo-racca-pa) [inline src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java:86]: correggiamo il typo addres con address

**Review by vincenzo-racca-pa — CHANGES_REQUESTED** [review]

File: src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java (hunk 1 above)
Comment (ariannazafarana) [reply src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java]: fixato

File: src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java (hunk 2 above)
Comment (ariannazafarana) [reply src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java]: fixato

File: src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java (hunk 3 above)
Comment (ariannazafarana) [reply src/main/java/it/pagopa/pn/service/desk/action/impl/ValidationOperationActionImpl.java]: fixato

**Review by vincenzo-racca-pa — APPROVED** [review]

**Pull request closed without merging**